packages/core/test/unit/patchers/promise_p.test.js

```diff
@@ -0,0 +1,152 @@
+  getNamespaceStub.returns({
+    bind: bindStub
+  });
+});
+
+beforeEach(() => {
+  sinon.restore();
+  Promise.prototype.then = origThen;
+  Promise.prototype.catch = origCatch;
+});
+
+after(() => {
+  sinon.restore();
+  Promise.prototype.then = origThen;
+  Promise.prototype.catch = origCatch;
+});
+
+function verifyBindings (segment, isAutomaticMode, onFulfilled, onRejected) {
+  if (segment && isAutomaticMode) {
+    if (onFulfilled) {
+      sinon.assert.calledWith(bindStub, onFulfilled);
+    }
+    if (onRejected) {
+      sinon.assert.calledWith(bindStub, onRejected);
+    }
+  } else {
+    sinon.assert.notCalled(bindStub);
+  }
+}
+
+function testThenOnFulfilled(segment, isAutomaticMode) {
+  const onFulfilled = result => {
+    expect(result).to.equal('resolved');
+    verifyBindings(segment, isAutomaticMode, onFulfilled);
+  };
+  return new Promise((resolve) => {
+    setTimeout(() => resolve('resolved'), 200);
+  }).then(onFulfilled);
+}
+
+function testThenOnRejected(segment, isAutomaticMode, hasOnFulfilled = true) {
+  const onFulfilled = hasOnFulfilled ? () => {
+    throw new Error('Not rejected');
+  } : undefined;
+
+  const onRejected = error => {
+    expect(error.message).to.equal('rejected');
+    verifyBindings(segment, isAutomaticMode, onFulfilled, onRejected);
+  };
+  return new Promise((_, reject) => {
+    setTimeout(() => reject(new Error('rejected')), 200);
+  }).then(onFulfilled, onRejected);
+}
+
+function testCatch(segment, isAutomaticMode) {
+  const onRejected = error => {
+    verifyBindings(segment, isAutomaticMode, undefined, onRejected);
+    expect(error.message, 'rejected');
+  };
+  return new Promise((_, reject) => {
+    setTimeout(() => reject(new Error('rejected')), 200);
+  }).catch(onRejected);
+}
+
+function createTests(segment, isAutomaticMode) {
+  it('Promise.prototype.then', async () => {
+    capturePromise();
+    expect(Promise.prototype.then).not.to.equal(origThen);
+    await testThenOnFulfilled(segment, isAutomaticMode);
+    await testThenOnRejected(segment, isAutomaticMode);
+    await testThenOnRejected(segment, isAutomaticMode, false);
+  });
+
+  it('Promise.prototype.catch', async () => {
+    capturePromise();
+    expect(Promise.prototype.catch).not.to.equal(origCatch);
+    await testCatch(segment, isAutomaticMode);
+  });
+}
+
+describe('capturePromise', () => {
+  describe('no segment', () => {
+    beforeEach(() => {
+      getSegmentStub.throws('No segment');
+      isAutomaticModeStub.returns(true);
+    });
+
+    createTests(false);
+  });
+
+  describe('not automaticMode', () => {
+    beforeEach(() => {
+      getSegmentStub.returns({});
+      isAutomaticModeStub.returns(false);
+    });
+
+    createTests(false);
+  });
+
+  describe('binds methods', () => {
+    beforeEach(() => {
+      getSegmentStub.returns({});
+      isAutomaticModeStub.returns(true);
+    });
+    createTests(true);
+  });
+});
+
+describe('capturePromise.patchThirdPartyPromise', () => {
```

Comment:
I'm probably missing something, but where is this `patchThirdPartyPromise` function defined?

Reply:
Oof well this was dumb, just didn't have the bottom 6 lines of `promise_p` loaded on the "files changed" view. This is good to go!